components/domains/folder/FolderButtonList.tsx

```diff
@@ -0,0 +1,112 @@
+import styled from "styled-components";
+import React from "react";
+import { getSelectData } from "../../../pages/api/FolderApi";
+import { Folder } from "../../../types/folder";
+import { Link } from "../../../types/common";
+
+function Button({
+  folder,
+  setSelectFolderLinks,
+  setId,
+}: {
+  folder: Folder;
+  setSelectFolderLinks: React.Dispatch<React.SetStateAction<Link[]>>;
+  setId: React.Dispatch<React.SetStateAction<number>>;
+}) {
+  const handleChangeID = async () => {
+    const { data } = await getSelectData(folder.id);
+    setSelectFolderLinks(data);
+    setId(folder.id);
+  };
+
+  return (
+    <StyledButton
+      onClick={() => {
+        handleChangeID();
+      }}
+    >
+      {folder && folder.name}
+    </StyledButton>
+  );
+}
+
+function FolderButtonList({
+  folderList,
+  setSelectFolderLinks,
+  setId,
+  openModal,
+}: {
+  folderList: Folder[];
+  setSelectFolderLinks: React.Dispatch<React.SetStateAction<Link[]>>;
+  setId: React.Dispatch<React.SetStateAction<number>>;
+  openModal: () => void;
+}) {
+  return (
+    <div>
+      <StyledButtonBox>
+        <StyledButtons>
+          {folderList?.map((folder) => (
+            <Button
+              folder={folder}
+              key={folder.id}
+              setSelectFolderLinks={setSelectFolderLinks}
+              setId={setId}
+            />
+          ))}
```

Comment:
setId 등을 더 내려주지않아도되겠습니다

```suggestion
          {folderList?.map((folder) => (
            <FolderBadge onClick={handleClick /** 이걸로 fetch하고 다함 */}>{folder.name}</FolderBadge>
          ))}
```